systemservices/ethwallet/x/xgo-ethereum/xaccounts/accounts.go

```diff
@@ -0,0 +1,23 @@
+package xaccounts
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// GetAccount retrieves an account from its address.
+func GetAccount(keystore *keystore.KeyStore, address string) (account accounts.Account, err error) {
+	err = fmt.Errorf("Account not found")
```

Comment:
```go
func GetAccount(keystore *keystore.KeyStore, address common.Address) (accounts.Account, error) {
	for _, account := range keystore.Accounts() {
		if account.Address == hexaddress {
			return account, nil
		}
	}
	return nil, fmt.Errorf("Account not found")
}
```

Plus you can't use a `HexToAddress` here because it internally [drops](https://github.com/ethereum/go-ethereum/blob/f91312dbdbb9e04ef578946226e5d8069d5dfd5a/common/bytes.go#L97) all errors